src/main/java/core/basesyntax/service/DataConverterImpl.java

```
@@ -0,0 +1,45 @@
+package core.basesyntax.service;
+
+import core.basesyntax.exception.FruitShopException;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.Operation;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    private static final String COMMA = ",";
+
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> rawLines) {
+        List<FruitTransaction> result = new ArrayList<>();
+        if (rawLines == null || rawLines.isEmpty()) {
+            return result;
```

This code returns an empty list when `rawLines` is null: `if (rawLines == null || rawLines.isEmpty()) { return result; }`. The checklist requires handling null parameters explicitly: "Null parameters" and the guideline: "Remember that your code will need to be tested, so try to anticipate and handle all invalid input data in advance." Instead of returning an empty list when `rawLines == null`, throw an informative FruitShopException mentioning the method and input.